src/fparser/two/symbol_table.py

```diff
@@ -701,6 +743,49 @@ def root(self):
             current = current.parent
         return current
 
+    @property
+    def wildcard_imports(self):
+        """
+        :returns: names of all modules with wildcard imports into this scope or an
+                  empty list if there are none.
+        :rtype: List[Optional[str]]
+        """
+        mod_names = set()
+        for mod_name, mod in self._modules.items():
+            if mod.wildcard_import:
+                mod_names.add(mod_name)
+        if self.parent:
+            # Any wildcard imports in a parent scope will affect this scoping
+            # region so carry on up. Note that if the root scoping region in
+            # the current file is a SUBMODULE then we will be missing whatever
+            # is brought into scope in the parent MODULE (since that will typically
+            # be in a separate source file).
+            mod_names.update(self.parent.wildcard_imports)
+
+        return sorted(list(mod_names))
+
+    @property
+    def all_symbols_resolved(self):
+        """
+        :returns: whether all symbols in this scope have been resolved. i.e. if
+            there are any wildcard imports or this table is within a submodule
+            then there could be symbols we don't have definitions for.
```

Comment:
If a variable is not used in a routine, do we actually care if it is resolved or not? I'm not saying we should change anything as I understand this is from the perspective of the symbol_table but we could potentially ignore/remove symbols that are declared but not used, or wildcard imports if all symbols that are used are already resolved.

Reply:
We could potentially remove symbols that aren't used although, as you say, this method is about whether or not there could be symbols being brought into scope from elsewhere. I tried to think of a better name for it but failed. However, I think the docstring is pretty clear.